pkg/importer/dataStream.go

```diff
@@ -905,14 +909,18 @@ func (d *DataStream) copy(dest string) error {
 			return err
 		}
 	} else {
-		if util.GetAvailableSpace(d.ScratchDataDir) <= int64(0) {
+		if d.VolumeMode != string(v1.PersistentVolumeBlock) && util.GetAvailableSpace(d.ScratchDataDir) <= int64(0) {
 			//Need scratch space but none provided.
 			return ErrRequiresScratchSpace
 		}
 		// Replace /data/target name with scratch path/target name
 		tmpDest := filepath.Join(d.ScratchDataDir, filepath.Base(dest))
 
-		err := StreamDataToFile(d.topReader(), tmpDest)
+		if d.VolumeMode == string(v1.PersistentVolumeBlock) {
```

Comment:
The temp dest should always be scratch space if available, we might optimize this for RAW/ISO files by skipping that step and writing directly to the destination, but this breaks scratch space and block devices.

Reply:
the use case I'm using for this phase, is importing ISO. This use case falls to the 'scratch space' block as it is not qemu. It should not use scratch space and tmpDest should not be /scratchSpace/.....
As we talk about this back and forth, and as my wish not to break you code, could you please write me the code that you want to see, and I will write it...

Reply:
Since you insist on not sending the ISO through the QEMU code path, you will have to add an additional code path like this:
```go
if d.isHTTPQcow2() {
  err := d.convertQcow2ToRawStream(dest)
  if err != nil {
    return err
  }
} else if isHTTPRaw() {
  klog.V(3).Infoln("Validating RAW file")
  err := qemuOperations.Validate(d.url.String(), d.AvailableSpace)
  if err != nil {
    return errors.Wrap(err, "Streaming RAW image validation failed")
  }
  err := StreamDataToFile(d.topReader(), dest)
  if err != nil {
    return errors.Wrap(err, "Streaming to raw failed")
  }
} else {
  ...scratchspacepath...
}
```
And add a method like this:
```go
func (d *DataStream) isHTTPRaw() bool {
  return (d.url.Scheme == "http" || d.url.Scheme == "https") &&
    d.AccessKey == "" &&
    d.SecKey == "" &&
    d.isIsoImage &&
    len(d.Readers) == 2
}
```
You are basically duplicating the QEMU code path for raw files. I would prefer you would just set qemu to true and be done, but since you don't want to do that, this will work.